File: Leetcode/210125/Java/GridGame.java
// Problem Link: https://leetcode.com/problems/grid-game/description/
// Solution Link: https://www.youtube.com/watch?v=Y8VC_OnkazE

class Solution {
    public long gridGame(int[][] grid) {
      long firstSum =0;

      for(int num: grid[0]){
         firstSum += num;
      }

      long secondSum =0;
      long miniSum = Long.MAX_VALUE;

      for(int ti=0; ti<grid[0].length; ti++){
          firstSum -= grid[0][ti];

          miniSum = Math.min(miniSum, Math.max(firstSum, secondSum));

          secondSum += grid[1][ti];
      }

      return miniSum; 
    }
}

File: Leetcode/210125/Python/GridGame.py
# Problem Link: https://leetcode.com/problems/grid-game/
# Solution Link: https://www.youtube.com/watch?v=Y8VC_OnkazE

class Solution(object):
    def gridGame(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        firstSum =0
        for i in range(len(grid[0])):
            firstSum += grid[0][i]
        secondSum =0
        minSum = 10000000000000000000001
        for i in range(len(grid[0])):
            firstSum -= grid[0][i]
            minSum = min(minSum, max(firstSum, secondSum))
            secondSum += grid[1][i]
        return minSum
